tests/openai/test_openai_protocol.py

```diff
@@ -0,0 +1,189 @@
+            temperature=-0.1,
+        )
+    assert "temperature" in str(exc_info.value)
+
+
+def completions_stream(openai_testing_model):  # noqa: F811
+    i = 0
+    for completion in client.completions.create(
+        model=openai_testing_model, prompt="Hello world", stream=True, top_p=1
+    ):
+        i += 1
+        assert completion
+        assert completion.id
+        assert isinstance(completion.choices, list)
+        assert isinstance(completion.choices[0].text, str)
+    assert i > 4
+
+
+def chat_stream(openai_testing_model):  # noqa: F811
+    i = 0
+    for chat_completion in client.chat.completions.create(
+        model=openai_testing_model,
+        messages=[{"role": "user", "content": "Hello world"}],
+        stream=True,
+        temperature=0.4,
+        frequency_penalty=0.02,
+        top_p=1,
+    ):
+        if i == 0:
+            assert chat_completion
+            assert chat_completion.id
+            assert isinstance(chat_completion.choices, list)
+            assert chat_completion.choices[0].delta.role
+        else:
+            assert chat_completion
+            assert chat_completion.id
+            assert isinstance(chat_completion.choices, list)
+            assert chat_completion.choices[0].delta == {} or hasattr(
+                chat_completion.choices[0].delta, "content"
+            )
+        i += 1
+    assert chat_completion
+    assert chat_completion.id
+    assert isinstance(chat_completion.choices, list)
+    assert chat_completion.choices[0].delta == {} or hasattr(
+        chat_completion.choices[0].delta, "content"
+    )
+    assert chat_completion.choices[0].model_fields["finish_reason"]
+    assert i > 4
+
+
+def completions_stream_bad_request(openai_testing_model):  # noqa: F811
+    with pytest.raises(openai.APIError) as exc_info:
+        for _ in client.completions.create(
+            model=openai_testing_model,
+            prompt="Hello world",
+            stream=True,
+            temperature=-0.1,
+        ):
+            pass
+    assert "temperature" in str(exc_info.value)
+
+
+def chat_stream_bad_request(openai_testing_model):  # noqa: F811
+    with pytest.raises(openai.APIError) as exc_info:
+        for _chat_completion in client.chat.completions.create(
+            model=openai_testing_model,
+            messages=[{"role": "user", "content": "Hello world"}],
+            stream=True,
+            temperature=-0.1,
+        ):
+            pass
+    assert "temperature" in str(exc_info.value)
+
+
+executed_models = {}
+
+
+# Parametrize the test function with different combinations of parameters
+@pytest.mark.parametrize(
+    "model,test_func",
+    [
+        (model, test_func)
+        for model in ["gpt2"]
+        for test_func in [
+            "models",
+            # "completions",
+            # "completions_stream",
+            "chat",
+            "chat_stream",
+            # "chat_bad_request",
+            # "chat_stream_bad_request"
```

Comment:
Can we enable these two starting with `chat_` and remove others starting with `completions_` ?

Reply:
I'm trying make them enable